scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -221,17 +222,35 @@ class FossId(
                         .checkResponse("create project")
                 }
 
-                log.info { "Creating scan for URL $url and revision $revision ..." }
+                val scans = service.listScansForProject(user, apiKey, projectCode)
+                    .checkResponse("list scans for project").data
+                requireNotNull(scans)
 
-                val scanCode = createScan(projectCode, url, revision)
-
-                log.info { "Initiating download from Git repository ..." }
+                val existingScan = scans.sortedByDescending { scan -> scan.id }.find { scan ->
+                    scan.gitBranch == revision && scan.gitRepoUrl == url
+                }
 
-                service.downloadFromGit(user, apiKey, scanCode)
-                    .checkResponse("download data from Git", false)
+                val scanCode = kotlin.run {
+                    if (existingScan == null) {
+                        log.info { "No scan found for url=$url and revision=$revision. Creating scan ..." }
```

Comment:
As from the syntax of a URL it's usually clear that it's a URL, I believe you can drop "url=", and in "revision=" please replace the "=" with " " to make a fluently readable sentence.